src/workerd/jsg/jsg.h

```diff
@@ -2063,6 +2107,68 @@ class Lock {
   ContextScope enterContextScope(v8::Local<v8::Context> context);
   // Ensures that we are currently within the scope of the given v8::Context
 
+  // ====================================================================================
+  JsValue undefined() KJ_WARN_UNUSED_RESULT;
+  JsValue null() KJ_WARN_UNUSED_RESULT;
+  JsBoolean boolean(bool val) KJ_WARN_UNUSED_RESULT;
+  JsNumber num(double) KJ_WARN_UNUSED_RESULT;
+  JsNumber num(float) KJ_WARN_UNUSED_RESULT;
+  JsInt32 num(int8_t) KJ_WARN_UNUSED_RESULT;
+  JsInt32 num(int16_t) KJ_WARN_UNUSED_RESULT;
+  JsInt32 num(int32_t) KJ_WARN_UNUSED_RESULT;
+  JsBigInt num(int64_t) KJ_WARN_UNUSED_RESULT;
+  JsUint32 num(uint8_t) KJ_WARN_UNUSED_RESULT;
+  JsUint32 num(uint16_t) KJ_WARN_UNUSED_RESULT;
+  JsUint32 num(uint32_t) KJ_WARN_UNUSED_RESULT;
+  JsBigInt num(uint64_t) KJ_WARN_UNUSED_RESULT;
```

Comment:
Hmm, I'd find it a bit surprising that depending on the size of the integer I pass in here, I might get a BigInt instead of a Number as the result. Since C++ is pretty liberal in implicitly converting integers to larger widths this might lead to getting a BigInt when not expected or desired.

I'd instead expect `num()` always produces a Number (or Int32 or Uint32 which are equivalent to Numbers, I think?) and instead have a separate `bigInt()` method I'd use explicitly when that's what I want. I'd expect it's my responsibility to make sure that the possible precision loss is acceptable.

Reply:
Definitely fair :-) ... I was debating this one a bit.

Reply:
Renamed!